bc/core/utils/status/base.py

```diff
@@ -69,13 +69,20 @@ def _check_output_validity(self, text: str) -> bool:
         """
         Checks whether the provided text exceeds the maximum allowed length.
 
+        Strips links from the output text since they use a fixed character
+        count.
+
         Args:
             text (str): The text to be evaluated.
 
         Returns:
             bool: True if the text length is within the limit, False otherwise.
         """
-        return len(text) <= self.max_characters
+        url_pattern = r"https?://\S+"
+        url_match = re.findall(url_pattern, text)
+        output = re.sub(url_pattern, "", text)
+
+        return len(output) + 23 * len(url_match) <= self.max_characters
```

Comment:
I found it weird that the length of URLs in tweets and toots is both 23, and sort of couldn't believe that this was meant to be in the BaseTemplate. 

Maybe a comment would help:

```suggestion
        url_pattern = r"https?://\S+"
        url_count = len(re.findall(url_pattern, text))
        linkless_output = re.sub(url_pattern, "", text)

        # Twitter and Mastodon both count links as 23 chars at present
        return len(linkless_output) + (23 * url_count) <= self.max_characters
```

(I did some minor refactoring too, mostly just renaming variables.)

Reply:
I'll add this comment. Thanks!